src/søknad/steg/4-barnasbosted/BarnasBosted.tsx

```diff
@@ -0,0 +1,197 @@
+import { Element } from 'nav-frontend-typografi';
+import { Input } from 'nav-frontend-skjema';
+import { Checkbox } from 'nav-frontend-skjema';
+import { ISpørsmål } from '../../../models/spørsmal';
+import {
+  borINorge,
+  avtaleOmDeltBosted,
+  harAnnenForelderSamværMedBarn,
+  harDereSkriftligSamværsavtale,
+  boddSammenFør,
+  borISammeHus,
+  hvorMyeSammen
+} from './ForeldreConfig';
+import { IForelder } from '../../../models/person';
+import JaNeiSpørsmål from '../../../components/spørsmål/JaNeiSpørsmål';
+import MultiSvarSpørsmål from '../../../components/spørsmål/MultiSvarSpørsmål';
+import KomponentGruppe from '../../../components/gruppe/KomponentGruppe';
+import BarnasBostedHeader from './BarnasBostedHeader';
+import FeltGruppe from '../../../components/gruppe/FeltGruppe';
+import { useIntl } from 'react-intl';
+import Datovelger, { DatoBegrensning } from '../../../components/dato/Datovelger';
+import LocaleTekst from '../../../language/LocaleTekst';
+import { AlertStripeInfo } from 'nav-frontend-alertstriper';
+import HvordanPraktiseresSamværet from './HvordanPraktiseresSamværet';
+
+const BarnasBosted: React.FC = () => {
+  const intl = useIntl();
+  const { søknad } = useSøknadContext();
+  const [forelder, settForelder] = useState<IForelder>({});
+
+  const settHarForelderSamværMedBarn = (spørsmål: string, valgtSvar: string) => {
+    const nyForelder = {...forelder, [harAnnenForelderSamværMedBarn.spørsmål_id]: valgtSvar};
+
+    if (valgtSvar === intl.formatMessage({ id: 'barnasbosted.spm.andreForelderenSamværNei'})) {
+      delete nyForelder.harDereSkriftligSamværsavtale;
+      delete nyForelder.hvordanPraktiseresSamværet;
+    }
+
+    settForelder(nyForelder);
+  }
+
+  const settHarDereSkriftligSamværsavtale = (spørsmål: string, valgtSvar: string) => {
+    const nyForelder = {...forelder, [harDereSkriftligSamværsavtale.spørsmål_id]: valgtSvar};
+
+    if (valgtSvar !== intl.formatMessage({ id: 'barnasbosted.spm.jaIkkeKonkreteTidspunkt'})) {
+      delete nyForelder.hvordanPraktiseresSamværet;
+    }
+
+    settForelder(nyForelder);
+  }
+
+  const settHarBoddsammenFør = (spørsmål: ISpørsmål, valgtSvar: boolean) => {
+    const nyForelder = {...forelder, [boddSammenFør.spørsmål_id]: valgtSvar};
+
+    if (valgtSvar === false) {
+      delete nyForelder.flyttetFra;
+    }
+
+    settForelder(nyForelder);
+  }
+
+  const visSamværsavtaleAdvarsel = (valgtSvar: string) => {
+    return valgtSvar === intl.formatMessage({ id: 'barnasbosted.spm.jaIkkeKonkreteTidspunkt' });
+  }
+
+  const visSkriftligSamværsavtaleSpørsmål = (svarAndreForelderenSamvær: string) => {
+    return svarAndreForelderenSamvær && svarAndreForelderenSamvær !== intl.formatMessage({ id: 'barnasbosted.spm.andreForelderenSamværNei' });
+  }
+
+  const visHvordanPraktiseresSamværet = (valgtSamværsrett: string) => {
+    return valgtSamværsrett === intl.formatMessage({ id: 'barnasbosted.spm.jaIkkeKonkreteTidspunkt' });
+  }
```

Comment:
Kunne man laget en felles funksjon for å sammenligne valgtSvar/variabel og tekstid? 
_________
**Diskusjon:**
På en annen side, så kan vi utvider søknadsfeltene våre til å lagre tekstidene til spørsmål og valgtsvar som `label_id `og` verdi_id` i context også: 
```  
label: string;
  label_id?: string;
  verdi: string;
  verdi_id?: string;
```
Og da kan vi kanskje unngå å måtte bruke `valgtSvar === intl.formatMessage({..})`for å sammenligne spørsmål sitt svarsalternativ med med det som er lagret i context. Hvis du setter `label_id` og `verdi_id` til optional for nå, så slipper du å "fikse" alle de andre stegene. Tanker? :) 

Jeg prøver å gjøre det i min branch også med arbeidssituasjon atm :)

Reply:
Kom på at det du nevnte på slack: 

> "Når det gjelder de tekst-idene, så endret jeg ikke de nå. Kan ta en større gjennomgang der sånn at det blir konsistent når vi har blitt ferdig med stegene"

Var det dette du mente? I såfall,  se bort fra den lange kommentaren her! Og, så er jeg enig :) Er greit å vente med det for å få litt fart på stegene.

Reply:
Jepp, mente både dette og navngivingen av id-ene :)